playground/src/components/details/editor/file_editor.tsx

```diff
@@ -0,0 +1,370 @@
+  createNewFeedback,
+}: FileEditorProps) {
+  const monaco = useMonaco();
+  const editorRef = useRef<editor.IStandaloneCodeEditor>();
+
+  // Only render feedbacks that are relevant to the current file
+  // Unreferenced feedbacks are always shown
+  const feedbacksToRender = feedbacks?.filter((feedback) => {
+    const referenceType = getFeedbackReferenceType(feedback);
+    if (referenceType === "unreferenced") {
+      return false;
+    }
+    if ("file_path" in feedback) {
+      return feedback.file_path === filePath;
+    } else {
+      return true;
+    }
+  });
+  const modelPath = (identifier ? `${identifier}/` : "") + (filePath ?? "default");
+
+  // Height used for autoHeight
+  const [height, setHeight] = useState<number>(18);
+  const [hoverPosition, setHoverPosition] = useState<Position | undefined>(undefined);
+  const [selection, setSelection] = useState<Selection | undefined>(undefined);
+  const [isMounted, setIsMounted] = useState(false);
+
+  // Using state for non-react handlers is not working so we use a ref
+  // This is a workaround to bridge the gap between react and monaco
+  const latestAddFeedbackState = useRef({
+    isPressed: false,
+    selection: selection,
+    hoverPosition: hoverPosition,
+    feedbacks: feedbacks,
+    filePath: filePath,
+  });
+  const addFeedbackHoverDecorations = useRef<string[]>([]);
+
+  // Called when the user presses the add feedback button
+  const addFeedbackPressed = (referenceType: FeedbackReferenceType) => {
+    const model = editorRef.current?.getModel();
+    if (!createNewFeedback || !onFeedbacksChange || !model) return;
+
+    // Get the current state (workaround for non-react handlers)
+    const { selection, hoverPosition, feedbacks, filePath } =
+      latestAddFeedbackState.current;
+
+    let newFeedback = createNewFeedback();
+    if (referenceType === "referenced") {
+      if (newFeedback.type === "text") {
+        if (selection && !selection.isEmpty()) {
+          const startIndex = model.getOffsetAt({
+            lineNumber: selection.startLineNumber,
+            column: selection.startColumn,
+          });
+          const endIndex = model.getOffsetAt({
+            lineNumber: selection.endLineNumber,
+            column: selection.endColumn,
+          });
+          newFeedback = {
+            ...newFeedback,
+            index_start: startIndex,
+            index_end: endIndex,
+          };
+        } else if (hoverPosition) {
+          const startIndex = model.getOffsetAt({
+            lineNumber: hoverPosition.lineNumber,
+            column: model.getLineMinColumn(hoverPosition.lineNumber),
+          });
+          const endIndex = model.getOffsetAt({
+            lineNumber: hoverPosition.lineNumber,
+            column: model.getLineMaxColumn(hoverPosition.lineNumber),
+          });
+          newFeedback = {
+            ...newFeedback,
+            index_start: startIndex,
+            index_end: endIndex,
+          };
```

Comment:
```suggestion
          newFeedback.index_start = model.getOffsetAt({
            lineNumber: hoverPosition.lineNumber,
            column: model.getLineMinColumn(hoverPosition.lineNumber),
          });
          newFeedback.index_end = model.getOffsetAt({
            lineNumber: hoverPosition.lineNumber,
            column: model.getLineMaxColumn(hoverPosition.lineNumber),
          });
```
I think this approach would also be more readable in the other places here

Reply:
I should do that more often 🤔 thanks!